tests/wrappers/test_multioutput.py

```diff
@@ -0,0 +1,48 @@
+from functools import partial
+
+import pytest
+import torch
+from sklearn.metrics import accuracy_score
+
+from torchmetrics.classification import Accuracy
+from torchmetrics.regression import R2Score
+from torchmetrics.wrappers.multioutput import MultioutputWrapper
+
+
+def _multioutput_sk_accuracy(preds, target, num_outputs):
+    accs = []
+    for i in range(num_outputs):
+        accs.append(accuracy_score(torch.argmax(preds[:, :, i], dim=1), target[:, i]))
+    return accs
+
+
+@pytest.mark.parametrize(
+    "metric, compare_metric, pred_generator, target_generator, num_rounds",
+    [
+        (
+            MultioutputWrapper(R2Score(), num_outputs=2),
+            R2Score(num_outputs=2, multioutput="raw_values"),
+            partial(torch.randn, 10, 2),
+            partial(torch.randn, 10, 2),
+            2,
+        ),
+        (
+            MultioutputWrapper(Accuracy(num_classes=3), num_outputs=2),
+            partial(_multioutput_sk_accuracy, num_outputs=2),
+            partial(torch.rand, 10, 3, 2),
+            partial(torch.randint, 3, (10, 2)),
+            2,
+        ),
+    ],
+)
+def test_multioutput_wrapper(metric, compare_metric, pred_generator, target_generator, num_rounds):
```

Comment:
Based on how things are parameterized, it looks like it can fit into our standard testing framework like this:
https://github.com/PyTorchLightning/metrics/blob/e6d998e11fc2b93286d62beaa70f6a47647d77eb/tests/regression/test_r2.py#L71-L84
It would be great if you could do this, because the wrapper should then also be tested automatically for cuda and multi device (ddp) support

Reply:
Awesome, thanks for the tip! Will do.

Reply:
So in trying to get this to work, I realized it actually won't work because `forward` calls `reset` but in my current setup (and the bootstrap wrapper's by the way), setting `self.metrics = nn.ModuleList` doesn't register the underlying metrics as part of the state. As a result, `forward` will reset the state meaning `compute()` will return the result from the last batch rather than over all batches.

One idea I had was trying to make `metrics` part of state, but that won't work if I continue to use an `nn.ModuleList`. Alternatively, I was wondering if you think it makes sense to try and make the wrapper's state the union of the underlying states (with indexes to differentiate). Other ideas also welcome!

Reply:
Ok going to override `forward` actually, this isn't too messy.